### ISSUE TITLE
Wireguard support

### ISSUE THREAD

**Issue by mitsu98** [gnuton/asuswrt-merlin.ng#190]
Is it possible to support wireguard for XT8?

```
	WireGuard-serverwg21: Initialising WireGuard VPN 'Server' Peer (wg21) on 10.50.1.1:51820 (# ZenWiFi_XT8 Server #1)
RTNETLINK answers: Operation not supported
Unable to modify interface: Protocol not supported
Unable to modify interface: Protocol not supported
Cannot find device "wg21"

	***ERROR Initialisation ABORTED - 'wg setconf wg21 /tmp/wg21.27333 (/opt/etc/wireguard.d/wg21.conf)' FAILED
```

**Comment by gnuton** [comment]
For people willing to try wireguard here is the 387 beta with it

RT-AX82U
https://output.circle-artifacts.com/output/job/e8ea3493-b0e4-40da-8fce-804b3ce9dfdf/artifacts/0/workspace/release/RT-AX82U_386.7_0_cferom_puresqubi.w

XT8
https://output.circle-artifacts.com/output/job/160da88a-a4aa-429c-8f45-bee5b1a57be9/artifacts/0/workspace/release/RT-AX95Q_386.7_0_cferom_puresqubi.w

if you need images for more routers please ping me.
If it works for you I can make it part of the next release

**Comment by vichaos** [comment]
> For people willing to try wireguard here is the 387 beta with the support for wireguard
> 
> RT-AX82U https://output.circle-artifacts.com/output/job/e8ea3493-b0e4-40da-8fce-804b3ce9dfdf/artifacts/0/workspace/release/RT-AX82U_386.7_0_cferom_puresqubi.w
> 
> XT8 https://output.circle-artifacts.com/output/job/160da88a-a4aa-429c-8f45-bee5b1a57be9/artifacts/0/workspace/release/RT-AX95Q_386.7_0_cferom_puresqubi.w
> 
> if you need images for more routers please ping me. If it works for you I can make it part of the next release
> 
> Please use https://github.com/MartineauUK/wireguard#wireguard-manager to test it. UI is not available yet

Yes, it works on XT8, however it consumes a lot of cpu resources while the maximum throughput is ~130Mbps which is about 25% less than InstantGuard.

<img width="2032" alt="Screen Shot 2565-07-01 at 03 36 45" src="https://user-images.githubusercontent.com/724055/176780561-8aeb183c-bbd3-4ec5-9338-796bb312f40c.png">

**Comment by trebleta** [comment]
Any chance for dsl ax82u version, or it included in the lastest beta

**Comment by gnuton** [comment]
@trebleta it should be part of the beta2

**Comment by gnuton** [comment]
@vichaos the high CPU usage can be the bottle neck and reason of the bad performance. have you manage to get aroud that?

**Comment by vichaos** [comment]
> @vichaos the high CPU usage can be the bottle neck and reason of the bad performance. have you manage to get aroud that?

I found that disable QoS and tuning the mtu for wireguard interface is kinda help. However this is really depend on what kind of internet connection and network you are using. 

For example, My condo is using IPV4 PPPoE 2Gbps/1Gbps fiber internet,so the QoS is not necessary at all. And the proper wireguard mtu can be calculated as below.

internet_mtu = 1492 
sizeof_ip4_hdr = 20 (or 40 for ip6)
sizeof_udp_hdr = 8
wireguard_overhead = 32

ip4_wireguard_mtu = internet_mtu - sizeof_ip4_hdr - sizeof_udp_hdr - wireguard_overhead

ip4_wireguard_mtu = 1492-20-8-32 = 1432.

**Comment by FishFloat** [comment]
Hi
I would love to try this beta for RT-AX88U, could you please give me a link if you got one?
When will the 387 be released?

Best regards
/M

**Comment by trebleta** [comment]
Your model is not support by this firmware.

**Comment by FishFloat** [comment]
> Your model is not support by this firmware.

What do you mean? I'm running 386.7 now and what you are saying is that from 387 my router will be unsupported?

/F

**Comment by trebleta** [comment]
You say you have rt-ax88u, this firmware is not for your model of router.

**Comment by FishFloat** [comment]
ok thanks, I thought it was a general Merlin-firmware for all Asus router. Sorry

**Comment by prooshani** [comment]
@gnuton 
Can I test wireguard version on TUF-AX5400??

**Comment by brunnels** [comment]
Can you please provide a new link/build for RT-AX82U?  Current link is broken.
